app/src/pages/SingleContent.spec.ts

```diff
@@ -0,0 +1,113 @@
+import "fake-indexeddb/auto";
+import { describe, it, expect, afterEach, beforeEach } from "vitest";
+import { mount, shallowMount } from "@vue/test-utils";
+import { setActivePinia, createPinia } from "pinia";
+import SingleContent from "./SingleContent.vue";
+import { mockPostDto, mockEnglishContentDto, mockCategoryContentDto } from "@/tests/mockData";
+import { db, type BaseDocumentDto } from "luminary-shared";
+import waitForExpect from "wait-for-expect";
+
+describe("SinglePost", () => {
+    beforeEach(() => {
+        db.docs.bulkPut([
+            mockPostDto,
+            mockEnglishContentDto,
+            mockCategoryContentDto,
+        ] as BaseDocumentDto[]);
+
+        setActivePinia(createPinia());
+    });
+
+    afterEach(() => {
+        db.docs.clear();
+    });
+
+    it("displays the loading when no content", async () => {
+        const wrapper = mount(SingleContent, {
+            props: {
+                slug: mockEnglishContentDto.slug,
+            },
+        });
+
+        expect(wrapper.findComponent({ name: "LoadingSpinner" }).exists()).toBe(true);
+    });
+
+    it("displays the image content", async () => {
+        await db.docs.update(mockEnglishContentDto._id, { image: "test-image.jpg" });
+
+        const wrapper = mount(SingleContent, {
+            props: {
+                slug: mockEnglishContentDto.slug,
+            },
+        });
+
+        await waitForExpect(() => {
+            expect(wrapper.html()).toContain("test-image.jpg");
+        });
+    });
+
+    it("displays the video content when defined", async () => {
```

Comment:
Description could change (for the sake of consistency) to "displays the content video when defined"